compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -396,34 +398,53 @@ impl<'a, 'tcx> AbstractConstBuilder<'a, 'tcx> {
             | ExprKind::Field { .. }
             | ExprKind::ConstBlock { .. }
             | ExprKind::Adt(_) => self.error(
+                Some(node.span),
+                "unsupported operation in generic constant, this may be supported in the future",
+            )?,
+            ExprKind::Pointer { .. } =>
+            // dont know if this is correct
+            {
+                self.error(Some(node.span), "pointers are not allowed in generic constants")?
+            }
+            ExprKind::Continue { .. } | ExprKind::Break { .. } | ExprKind::Loop { .. } => self
+                .error(
                     Some(node.span),
-                    "unsupported operation in generic constant, this may be supported in the future",
+                    "loops and loop control flow are not supported in generic constants",
                 )?,
+            ExprKind::Box { .. } =>
+            // allocations not allowed in constants
+            {
+                self.error(Some(node.span), "allocations are not allowed in generic constants")?
+            }
+
+            // we handle valid unary/binary ops above
+            ExprKind::Unary { .. } | ExprKind::Binary { .. } | ExprKind::LogicalOp { .. } => {
+                self.error(Some(node.span), "unsupported operation in generic constants")?
```

Comment:
The reason we disallow `ExprKind::LogicalOp` is because `&&` and `||` are short circuiting and so imply controlflow. I think we should mention this. I'm not sure why we disallow `Binop::Offset` ``/// The `ptr.offset` operator`` (or what it is)

Reply:
I've added a msg for LogicalOps but nothing else